### PR TITLE
Update of FactionCreateCommand & add FactionSubCommand

### PR DESCRIPTION
**Pull request by DebHugo**
Now, you can do /f create {name}

### CODE REVIEW

**Comment by DebHugo** [comment]
I also just added the ability to see extension commands to /f help

File: resources/lang/en_EN.yml [hunk 1]
```diff
@@ -176,26 +176,25 @@ CANT_INTERACT_SPAWN: " §c>> §4You can't interact with that in the spawn"
...
-COMMAND_HELP_DESCRIPTION: "Displays the orders"
-COMMAND_INFO_DESCRIPTION_GLOBAL: "Displays information about a faction"
-COMMAND_CLAIM_INFO_DESCRIPTION: "Displays information about a chunk"
-COMMAND_EXTENSION_DESCRIPTION: "Display extensions enabled"
-COMMAND_SCOREBOARD_DESCRIPTION: "Place a leaderboard to your location"
-COMMAND_SCOREBOARD_REMOVE_DESCRIPTION: "Remove the nearest leaderboard around you"
-COMMAND_TOP_DESCRIPTION: "Open the correct menu for the leaderboard"
-COMMAND_ADD_FLAG: "Add a claim considered like an area"
-COMMAND_REMOVE_FLAG: "Remove a claim considered like an area"
+COMMAND_FACTION: "Opens the main menu of FactionMaster"
```
Comment (ShockedPlot7560) [inline resources/lang/en_EN.yml]: I think it is not necessary to remove the ``_DESCRIPTION ``
This allows for better clarity and could be confusing with other translations

File: resources/lang/en_EN.yml [hunk 2]
```diff
@@ -176,26 +176,25 @@ CANT_INTERACT_SPAWN: " §c>> §4You can't interact with that in the spawn"
...
+COMMAND_HOME: "Opens the home menu"
+COMMAND_CREATE: "Opens the menu to create a faction"
+COMMAND_MAP: "Displays the map listing all claims"
+COMMAND_HELP: "Displays the orders"
+COMMAND_INFO: "Displays information about a faction"
+COMMAND_CLAIMINFO: "Displays information about a chunk"
+COMMAND_EXTENSION: "Display extensions enabled"
+COMMAND_PLACELEADERBOARD: "Place a leaderboard to your location"
+COMMAND_REMOVELEADERBOARD: "Remove the nearest leaderboard around you"
+COMMAND_ADDFLAG: "Add a claim considered like an area"
```
Comment (ShockedPlot7560) [inline resources/lang/en_EN.yml]: Good idea to group it, however, as said earlier, add the ``_DESCRIPTION``, to follow the same format

File: resources/lang/en_EN.yml [hunk 3]
```diff
@@ -176,26 +176,25 @@ CANT_INTERACT_SPAWN: " §c>> §4You can't interact with that in the spawn"
...
+COMMAND_REMOVEFLAG: "Remove a claim considered like an area"
```
Comment (ShockedPlot7560) [inline resources/lang/en_EN.yml]: Same as above

File: resources/lang/es_SPA.yml [hunk 4]
```diff
@@ -166,22 +166,22 @@ CANT_INTERACT_CLAIM: " §c>> §4No puedes interactuar con eso en territorio enem
...
+COMMAND_DELHOME: "Elimina la casa de la facción"
+COMMAND_TP: "Teletransportarse en la casa de la facción"
+COMMAND_HOME: "Abre el menú de inicio"
+COMMAND_CREATE: "Abre el menú para crear una facción"
+COMMAND_MAP: "Muestra el mapa que enumera todos los reclamos"
+COMMAND_HELP: "Muestra los pedidos"
+COMMAND_INFO: "Muestra información sobre una facción."
+COMMAND_CLAIMINFO: "Muestra información sobre un territorio"
+COMMAND_EXTENSION: "Extensiones de display habilitadas"
+COMMAND_PLACELEADERBOARD: "Mueve/habilita la tabla de clasificación y da los datos que se deben ingresar para guardarlos"
```
Comment (ShockedPlot7560) [inline resources/lang/es_SPA.yml]: Same as english file, add the ``_DESCRIPTION``

File: resources/lang/fr_FR.yml [hunk 5]
```diff
@@ -178,26 +178,25 @@ CANT_INTERACT_SPAWN: " §c>> §4Tu ne peux pas intéragir avec ça dans le spawn
...
+COMMAND_CREATE: "Ouvre le menu pour créer une faction"
+COMMAND_MAP: "Affiche la carte listant tout les claim"
+COMMAND_HELP: "Affiche les commandes"
+COMMAND_INFO: "Affiche les informations sur une faction"
+COMMAND_CLAIMINFO: "Affiche les informations sur un chunk"
+COMMAND_EXTENSION: "Affiche la liste des extensions actives"
+COMMAND_PLACELEADERBOARD: "Ajoute un panneaux des scores à vos pieds"
+COMMAND_REMOVELEADERBOARD: "Retire le panneau des scores le plus près de vous"
+COMMAND_ADDFLAG: "Ajoute un claim considéré comme une zone"
+COMMAND_REMOVEFLAG: "Retire un claim considéré comme une zone"
```
Comment (ShockedPlot7560) [inline resources/lang/fr_FR.yml]: Same as english

File: resources/lang/en_EN.yml (hunk 2 above)
Comment (ShockedPlot7560) [reply resources/lang/en_EN.yml]: as I just saw, potentially do the same with ``COMMAND_SYNCHRO``

File: resources/lang/pt_BR.yml [hunk 6]
```diff
@@ -172,25 +172,25 @@ CANT_INTERACT_SPAWN: " §c>> §4Você não pode interagir com isso no spawn"
...
+COMMAND_CREATE: "Abre o menu para criar uma facção"
+COMMAND_MAP: "Exibe o mapa listando todas as reivindicações"
+COMMAND_HELP: "Mostra os pedidos"
+COMMAND_INFO: "Exibe informações sobre uma facção"
+COMMAND_CLAIMINFO: "Exibe informações sobre um terreno"
+COMMAND_EXTENSION: "Extensões de exibição ativadas"
+COMMAND_PLACELEADERBOARD: "Coloque uma tabela de classificação para a sua localização"
+COMMAND_REMOVELEADERBOARD: "Remova a tabela de classificação mais próxima ao seu redor"
+COMMAND_ADDFLAG: "Remova a tabela de classificação mais próxima ao seu redor"
+COMMAND_REMOVEFLAG: "Remova uma reivindicação considerada como uma área"
```
Comment (ShockedPlot7560) [inline resources/lang/pt_BR.yml]: Same as english

File: resources/lang/tr_TR.yml [hunk 7]
```diff
@@ -172,24 +172,24 @@ CANT_INTERACT_SPAWN: " §c>> §4Bir spawn bölgesinde bununla etkileşime gireme
...
+COMMAND_HOME: "Ev menüyü açar"
+COMMAND_CREATE: "Bir klan oluşturmak için menüyü açar"
+COMMAND_MAP: "Tüm talepleri listeleyen haritayı görüntüler"
+COMMAND_HELP: "Yardım sayfasını açar"
+COMMAND_INFO: "Bir klan hakkında bilgi görüntüler"
+COMMAND_CLAIM: "Bir chunk hakkında bilgi görüntüler"
+COMMAND_EXTENSION: "Görüntü uzantıları etkinleştirir"
+COMMAND_PLACELEADERBOARD: "Lider tablonuzu taşıyın/etkinleştirin ve kaydetmek için girilecek verileri verir"
+COMMAND_ADDFLAG: "Bölge olarak kabul edilen araziyi ekler"
+COMMAND_REMOVEFLAG: "Bölge olarak kabul edilen bir araziyi kaldırır"
```
Comment (ShockedPlot7560) [inline resources/lang/tr_TR.yml]: Same as english

File: src/Command/Subcommand/FactionCreateCommand.php [hunk 8]
```diff
@@ -51,7 +54,13 @@ public function onRun(CommandSender $sender, string $aliasUsed, array $args): vo
 
 		$userEntity = MainAPI::getUser($sender->getName());
 		if ($userEntity->getFactionName() === null) {
-			Utils::processMenu(RouterFactory::get(RouteSlug::CREAFT_FACTION_ROUTE), $sender);
+		  if(!isset($args['name'])) {
+			  Utils::processMenu(RouterFactory::get(RouteSlug::CREAFT_FACTION_ROUTE), $sender);
+		  } else {
+		    $route = RouterFactory::get(RouteSlug::CREAFT_FACTION_ROUTE);
+		    $callable = $route->call();
+		    $callable($sender, $args['name']);
```
Comment (ShockedPlot7560) [inline src/Command/Subcommand/FactionCreateCommand.php]: You need to give an array of 2 elements.
```
[ null, factionName ]
```

File: src/Command/Subcommand/HelpCommand.php [hunk 9]
```diff
@@ -45,37 +45,13 @@ protected function prepare(): void {
 
 	public function onRun(CommandSender $sender, string $aliasUsed, array $args): void {
 		$this->player = $sender;
+
```
Comment (ShockedPlot7560) [inline src/Command/Subcommand/HelpCommand.php]: unnecessary change

File: src/Command/Subcommand/HelpCommand.php [hunk 10]
```diff
@@ -45,37 +45,13 @@ protected function prepare(): void {
...
-		}
-		if ($sender->hasPermission("factionmaster.leaderboard.place")) {
-			$sender->sendMessage($this->getString("/f removeleaderboard", "COMMAND_SCOREBOARD_REMOVE_DESCRIPTION"));
-		}
-		if ($sender->hasPermission("factionmaster.synchro.launch")) {
-			$sender->sendMessage($this->getString("/f synchro", "COMMAND_SYNCHRO"));
+		$sender->sendMessage($this->getString("/f", "COMMAND_FACTION"));
+		foreach (CommandManager::getCommands() as $command) {
+		  if ($command->testPermissionSilent($sender)) {
+			  $sender->sendMessage($this->getString("/f " .$command->getName(). " " .$command->getUsageMessage(), "COMMAND_" .strtoupper($command->getName())));
```
Comment (ShockedPlot7560) [inline src/Command/Subcommand/HelpCommand.php]: Cohesion in the language file is necessary in this case, or allow commands to provide their translation index.

File: src/Manager/CommandManager.php [hunk 11]
```diff
@@ -62,25 +62,25 @@ class CommandManager {
...
-		self::registerCommand(new HelpCommand("help", Utils::getText("", "COMMAND_HELP_DESCRIPTION")));
-		self::registerCommand(new ExtensionCommand("extension", Utils::getText("", "COMMAND_EXTENSION_DESCRIPTION")));
-		self::registerCommand(new PlaceLeaderboard("placeleaderboard", Utils::getText("", "COMMAND_SCOREBOARD_DESCRIPTION")));
-		self::registerCommand(new RemoveNearLeaderboardCommand("removeleaderboard", Utils::getText("", "COMMAND_SCOREBOARD_DESCRIPTION")));
-		self::registerCommand(new AddFlagCommand("addflag", Utils::getText("", "COMMAND_ADD_FLAG")));
-		self::registerCommand(new RemoveFlagCommand("removeflag", Utils::getText("", "COMMAND_REMOVE_FLAG")));
+		self::registerCommand(new HelpCommand("help", Utils::getText("", "COMMAND_HELP")));
+		self::registerCommand(new ExtensionCommand("extension", Utils::getText("", "COMMAND_EXTENSION")));
+		self::registerCommand(new PlaceLeaderboard("placeleaderboard", Utils::getText("", "COMMAND_PLACELEADERBOARD")));
+		self::registerCommand(new RemoveNearLeaderboardCommand("removeleaderboard", Utils::getText("", "COMMAND_REMOVELEADERBOARD")));
```
Comment (ShockedPlot7560) [inline src/Manager/CommandManager.php]: Change to follow the language file, with the ``_DESCRIPTION``

**Review by ShockedPlot7560 — CHANGES_REQUESTED** [review]
The logic is there indeed. However changes are necessary to keep the format of the command description: ``COMMAND_XXX_DESCRIPTION``.

For the help command, for me, the best solution is to provide the commands with a way to give their translation index, thus allowing to perform the foreach.

Thanks anyway, if you need help, ask for it

**Comment by DebHugo** [comment]
That's it, now everything is good ?

File: src/Command/Subcommand/HelpCommand.php [hunk 12]
```diff
@@ -38,44 +38,20 @@
...
-			$sender->sendMessage($this->getString("/f placeleaderboard <:slug>", "COMMAND_SCOREBOARD_DESCRIPTION"));
-		}
-		if ($sender->hasPermission("factionmaster.leaderboard.place")) {
-			$sender->sendMessage($this->getString("/f removeleaderboard", "COMMAND_SCOREBOARD_REMOVE_DESCRIPTION"));
-		}
-		if ($sender->hasPermission("factionmaster.synchro.launch")) {
-			$sender->sendMessage($this->getString("/f synchro", "COMMAND_SYNCHRO"));
+		foreach (CommandManager::getCommands() as $command) {
+		  if ($command->testPermissionSilent($sender)) {
+			  $sender->sendMessage($this->getString("/f " .$command->getName(). " " .$command->getUsageMessage(), $command->id));
```
Comment (ShockedPlot7560) [inline src/Command/Subcommand/HelpCommand.php]: ``$command->id`` will crash, id is a constant

File: src/Command/Subcommand/FactionSubCommand.php [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+  public const ID = "COMMAND_UNKNOW_DESCRIPTION";
+  
+  public function __construct(string $name, string $description = "", array $aliases = []) {
+    if($description === "") {
+      $description = Utils::getText("", static::ID);
+    }
+    parent::__construct($name, $description, $aliases);
+  }
+  
+  abstract public function onRun(CommandSender $sender, string $aliasUsed, array $args): void;
```
Comment (ShockedPlot7560) [inline src/Command/Subcommand/FactionSubCommand.php]: useless

**Review by ShockedPlot7560 — CHANGES_REQUESTED** [review]
Please stick to the code of conduct for the style of the files.

Several changes still to be made, in particular the fact of privileging a dedicated function rather than a constant for the ID

**Comment by DebHugo** [comment]
I fix but where I do not respect the code of conduct for the style of the files.

**Comment by ShockedPlot7560** [comment]
> I fix but where I do not respect the code of conduct for the style of the files.

Use php-cs-fixer if you are not sure, but mainly, at the indentation level

**Review by ShockedPlot7560 — CHANGES_REQUESTED** [review]
Set ``$id`` private and provide a getter to access them

**Comment by DebHugo** [comment]
Here, I tried to fix syntaxes as well as possible and I have a question: do you have discord? if so can you send me your tag.

**Comment by ShockedPlot7560** [comment]
> Here, I tried to fix syntaxes as well as possible and I have a question: do you have discord? if so can you send me your tag.

https://discord.gg/FrjX6WEpbD

File: src/Command/Subcommand/HelpCommand.php [hunk 14]
```diff
@@ -38,44 +38,20 @@
 use function str_replace;
 
 class HelpCommand extends BaseSubCommand {
```
Comment (ShockedPlot7560) [inline src/Command/Subcommand/HelpCommand.php]: Need to move ``BaseSubCommand`` to ``FactionSubCommand``

File: src/Command/Subcommand/HelpCommand.php [hunk 15]
```diff
@@ -38,44 +38,20 @@
...
-		}
-		if ($sender->hasPermission("factionmaster.leaderboard.place")) {
-			$sender->sendMessage($this->getString("/f placeleaderboard <:slug>", "COMMAND_SCOREBOARD_DESCRIPTION"));
-		}
-		if ($sender->hasPermission("factionmaster.leaderboard.place")) {
-			$sender->sendMessage($this->getString("/f removeleaderboard", "COMMAND_SCOREBOARD_REMOVE_DESCRIPTION"));
-		}
-		if ($sender->hasPermission("factionmaster.synchro.launch")) {
-			$sender->sendMessage($this->getString("/f synchro", "COMMAND_SYNCHRO"));
+		foreach (CommandManager::getCommands() as $command) {
```
Comment (ShockedPlot7560) [inline src/Command/Subcommand/HelpCommand.php:53]: You need to import CommandManager

File: src/Command/Subcommand/HelpCommand.php [hunk 16]
```diff
@@ -38,44 +38,20 @@
...
+			if ($command->testPermissionSilent($sender)) {
+				$sender->sendMessage($this->getString("/f " .$command->getName(). " " .$command->getUsageMessage(), $command->id));
```
Comment (ShockedPlot7560) [inline src/Command/Subcommand/HelpCommand.php]: Use ``$command->getId()`` instead of accessing a private properties.

File: src/Command/Subcommand/HomeTpCommand.php [hunk 17]
```diff
@@ -61,7 +63,7 @@ public function onRun(CommandSender $sender, string $aliasUsed, array $args): vo
 		if (Utils::haveAccess($permissions, $userEntity, PermissionIds::PERMISSION_TP_FACTION_HOME)) {
 			$home = MainAPI::getFactionHome($userEntity->getFactionName(), $args["name"]);
 			if ($home !== null) {
-				$sender->teleport(new Vector3($home->getX(), $home->getY(), $home->getZ()));
+				$sender->teleport($home->getVector());
```
Comment (ShockedPlot7560) [inline src/Command/Subcommand/HomeTpCommand.php]: Unrelated change, to be issued in another RP if needed

File: src/Command/Subcommand/SettingsCommand.php [hunk 18]
```diff
@@ -33,13 +33,12 @@
...
-use ShockedPlot7560\FactionMaster\libs\CortexPE\Commando\BaseSubCommand;
 use ShockedPlot7560\FactionMaster\Manager\ImageManager;
 use ShockedPlot7560\FactionMaster\Utils\Utils;
 
-class SettingsCommand extends BaseSubCommand {
-	protected function prepare(): void {
-	}
+class SettingsCommand extends FactionSubCommand {
+
+	private $id = "COMMAND_SETTINGS_DESCRIPTION";
```
Comment (ShockedPlot7560) [inline src/Command/Subcommand/SettingsCommand.php]: Unknown translation in files

File: src/Command/Subcommand/FactionSubCommand.php [hunk 19]
```diff
@@ -0,0 +1,58 @@
...
+use pocketmine\command\CommandSender;
+use ShockedPlot7560\FactionMaster\libs\CortexPE\Commando\BaseSubCommand;
+use ShockedPlot7560\FactionMaster\Utils\Utils;
+
+abstract class FactionSubCommand extends BaseSubCommand {
+
+	/**
+	 * @return string
+	 */
+	private $id = "COMMAND_UNKNOW_DESCRIPTION";
```
Comment (ShockedPlot7560) [inline src/Command/Subcommand/FactionSubCommand.php]: Unknown translation in files

Comment (ShockedPlot7560) [reply src/Command/Subcommand/FactionSubCommand.php]: You define this private property, it should be moved in visibility protect

File: src/Command/Subcommand/FactionSubCommand.php [hunk 20]
```diff
@@ -0,0 +1,58 @@
...
+
+	public function __construct(string $name, string $description = "", array $aliases = []) {
+		if($description === "") {
+			$description = Utils::getText("", $this->id);
```
Comment (ShockedPlot7560) [inline src/Command/Subcommand/FactionSubCommand.php]: Use ``$command->getId()`` instead of access to id directly

File: src/Manager/CommandManager.php [hunk 21]
```diff
@@ -52,40 +52,40 @@
...
+			self::registerCommand(new ClaimCommand("claim"));
+			self::registerCommand(new UnclaimCommand("unclaim"));
+			self::registerCommand(new SethomeCommand("sethome"));
+			self::registerCommand(new DelhomeCommand("delhome"));
+			self::registerCommand(new HomeTpCommand("tp"));
+			self::registerCommand(new HomeCommand("home"));
+			self::registerCommand(new MapCommand("map"));
+			self::registerCommand(new InfoCommand("info"));
+			self::registerCommand(new ClaimInfoCommand("claiminfo"));
 			self::registerCommand(new FactionTopCommand("top", "Give all the FactionMaster settings"));
```
Comment (ShockedPlot7560) [inline src/Manager/CommandManager.php]: Update this definition to track the translation file

File: src/Manager/ConfigManager.php [hunk 22]
```diff
@@ -45,12 +45,11 @@ class ConfigManager {
 	const LEVEL_VERSION = 0;
 	const TRANSLATION_VERSION = 2;
 	const LANG_FILE_VERSION = [
-		"en_EN" => 7,
-		"fr_FR" => 5,
-		"es_SPA" => 2,
-		"tr_TR" => 1,
-		"pt_BR" => 1,
-		"tr_TR" => 1
+		"en_EN" => 8,
```
Comment (ShockedPlot7560) [inline src/Manager/ConfigManager.php:48]: Why change the translation versions when you don't modify the files anymore?

**Review by ShockedPlot7560 — CHANGES_REQUESTED** [review]
A lot of changes, want to pay attention to everything. 
Instead of defining an ``id`` property to each class, you might as well transfer this directly to the ``getId`` function

File: src/Manager/ConfigManager.php (hunk 22 above)
Comment (DebHugo) [reply src/Manager/ConfigManager.php:48]: I forgot to put the old file back

**Comment by DebHugo** [comment]
Finish

File: src/Command/Subcommand/HelpCommand.php [hunk 23]
```diff
@@ -33,49 +33,25 @@
 namespace ShockedPlot7560\FactionMaster\Command\Subcommand;
 
 use pocketmine\command\CommandSender;
-use ShockedPlot7560\FactionMaster\libs\CortexPE\Commando\BaseSubCommand;
 use ShockedPlot7560\FactionMaster\Utils\Utils;
+namespace ShockedPlot7560\FactionMaster\Manager\CommandManager;
```
Comment (ShockedPlot7560) [inline src/Command/Subcommand/HelpCommand.php]: ``namespace`` sure ?

File: src/Command/Subcommand/FactionSubCommand.php [hunk 24]
```diff
@@ -0,0 +1,58 @@
...
+use pocketmine\command\CommandSender;
+use ShockedPlot7560\FactionMaster\libs\CortexPE\Commando\BaseSubCommand;
+use ShockedPlot7560\FactionMaster\Utils\Utils;
+
+abstract class FactionSubCommand extends BaseSubCommand {
+
+	/**
+	 * @return string
+	 */
+	protected $id = "COMMAND_UNKNOW_DESCRIPTION";
```
Comment (ShockedPlot7560) [inline src/Command/Subcommand/FactionSubCommand.php]: typing this properties, no need of PHPdoc

File: src/Manager/CommandManager.php [hunk 25]
```diff
@@ -52,40 +52,40 @@
...
-		self::registerCommand(new RemoveFlagCommand("removeflag", Utils::getText("", "COMMAND_REMOVE_FLAG")));
-		self::registerCommand(new SynchroCommand("synchro", Utils::getText("", "COMMAND_SYNCHRO")));
+		self::registerCommand(new HelpCommand("help"));
+		self::registerCommand(new ExtensionCommand("extension"));
+		self::registerCommand(new PlaceLeaderboard("placeleaderboard"));
+		self::registerCommand(new RemoveNearLeaderboardCommand("removeleaderboard"));
+		self::registerCommand(new AddFlagCommand("addflag"));
+		self::registerCommand(new RemoveFlagCommand("removeflag"));
+		self::registerCommand(new SynchroCommand("synchro"));
 		self::registerCommand(new SettingsCommand("settings", "Give all the FactionMaster settings"));
```
Comment (ShockedPlot7560) [inline src/Manager/CommandManager.php]: Use the same logic here, add a new index of translation in the file

File: src/Manager/ConfigManager.php [hunk 26]
```diff
@@ -50,7 +50,6 @@ class ConfigManager {
 		"es_SPA" => 2,
 		"tr_TR" => 1,
 		"pt_BR" => 1,
-		"tr_TR" => 1
```
Comment (ShockedPlot7560) [inline src/Manager/ConfigManager.php]: Useless change here, to be done later, out of this PR

**Review by ShockedPlot7560 — CHANGES_REQUESTED** [review]
I think it would be better, instead of putting a protected property in the abstract, to put ``getId``abstract and ask each command to fill it directly, without passing through a variable

File: src/Manager/CommandManager.php (hunk 25 above)
Comment (DebHugo) [reply src/Manager/CommandManager.php]: I can't because this is not translated into all languages

Comment (ShockedPlot7560) [reply src/Manager/CommandManager.php]: add it in en_EN and it will translate after

File: src/Command/Subcommand/InfoCommand.php [hunk 27]
```diff
@@ -36,14 +36,18 @@
...
 use ShockedPlot7560\FactionMaster\Utils\Utils;
 use function count;
 use function floor;
 use function strlen;
 
-class InfoCommand extends BaseSubCommand {
+class InfoCommand extends FactionSubCommand {
+
+	public function getId(): string {
+		return = "COMMAND_INFO_DESCRIPTION_GLOBAL";
```
Comment (ShockedPlot7560) [inline src/Command/Subcommand/InfoCommand.php]: Syntax error

File: src/Command/Subcommand/PlaceLeaderboard.php [hunk 28]
```diff
@@ -45,7 +44,12 @@
 use function implode;
 use function join;
 
-class PlaceLeaderboard extends BaseSubCommand {
+class PlaceLeaderboard extends FactionSubCommand {
+
+	public function getId(): string {
+		return = "COMMAND_SCOREBOARD_DESCRIPTION";
```
Comment (ShockedPlot7560) [inline src/Command/Subcommand/PlaceLeaderboard.php]: Syntax error

File: src/Command/Subcommand/HelpCommand.php [hunk 29]
```diff
@@ -33,49 +33,27 @@
...
-			$sender->sendMessage($this->getString("/f placeleaderboard <:slug>", "COMMAND_SCOREBOARD_DESCRIPTION"));
-		}
-		if ($sender->hasPermission("factionmaster.leaderboard.place")) {
-			$sender->sendMessage($this->getString("/f removeleaderboard", "COMMAND_SCOREBOARD_REMOVE_DESCRIPTION"));
-		}
-		if ($sender->hasPermission("factionmaster.synchro.launch")) {
-			$sender->sendMessage($this->getString("/f synchro", "COMMAND_SYNCHRO"));
+		foreach (CommandManager::getCommands() as $command) {
+			if ($command->testPermissionSilent($sender)) {
+				$sender->sendMessage($this->getString("/f " .$command->getName(). " " .$command->getUsageMessage(), $command->getId()));
```
Comment (ShockedPlot7560) [inline src/Command/Subcommand/HelpCommand.php]: You don't need to specify ``$command->getName()`` because it is included in the usageMessage

**Review by ShockedPlot7560 — CHANGES_REQUESTED** [review]

**Review by ShockedPlot7560 — APPROVED** [review]

**Comment by DebHugo** [comment]
It's good ?

**Comment by ShockedPlot7560** [comment]
After some testing, it will be ready to be merged yes

**Comment by DebHugo** [comment]
Ok
